service/controller/appcatalog/v1/resource/index/resource.go

```diff
@@ -0,0 +1,80 @@
+package index
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "indexv1"
```

Comment:
What is an index resource?

Reply:
Helm catalogs natively support an index.yaml. We want to use this in the front end as the master copy for metadata about App charts. 

e.g. https://github.com/giantswarm/sample-catalog/blob/master/index.yaml

So this resource will save the index to a configmap. This is to decouple the APIs from accessing GitHub directly. In the future its likely we will have either chartmuseum or harbor running in the control plane. But IMO its still better for the APIs to access via a configmap.

https://github.com/helm/chartmuseum
https://goharbor.io/

Note: Harbor still runs chartmuseum under the hood.